server/test/services/withings/lib/withings.poll.test.js

```diff
@@ -0,0 +1,406 @@
+        case OAUTH2.VARIABLE.GRANT_TYPE:
+          return 'authorization_code';
+        case OAUTH2.VARIABLE.INTEGRATION_SCOPE:
+          return 'user.info,user.metrics,user.activity,user.sleepevents';
+        case OAUTH2.VARIABLE.ACCESS_TOKEN:
+          return (
+            '{' +
+            '"access_token":"b96a86b654acb01c2aeb4d5a39f10ff9c964f8e4",' +
+            '"expires_in":10800,' +
+            '"token_type":"Bearer",' +
+            '"scope":"user.info,user.metrics,user.activity,user.sleepevents",' +
+            '"refresh_token":"11757dc7fd8d25889f5edfd373d1f525f53d4942",' +
+            '"userid":"33669966",' +
+            '"expires_at":"2020-11-13T20:46:50.042Z"' +
+            '}'
+          );
+        case OAUTH2.VARIABLE.CLIENT_ID:
+          countGetValueCallClientId += 1;
+          return 'fake_client_id';
+        default:
+          return '';
+      }
+    },
+    setValue: fake.resolves(null),
+    destroy: fake.returns(null),
+  },
+  device: {
+    create: fake.resolves(null),
+    destroyByServiceId: fake.resolves(null),
+    saveHistoricalState: fake.resolves(null),
+  },
+  user: {
+    get: fake.resolves([{ id: '0cd30aef-9c4e-4a23-88e3-3547971296e5' }]),
+  },
+  event: { emit: fake.returns(null) },
+};
+
+describe('WithingsHandler poll', () => {
+  const withingsHandler = new WithingsHandler(
+    gladys,
+    '55f177d7-bc35-4560-a1f0-4c58b9e9f2c4',
+    'http://localhost:9192',
+    'test',
+  );
+
+  before((done) => {
+    server.on({
+      method: 'GET',
+      path: '/v2/user',
+      reply: {
+        status: 200,
+        headers: { 'content-type': 'application/json' },
+        body: JSON.stringify({
+          status: 0,
+          body: {
+            devices: [
+              {
+                type: 'string',
+                model: 'string',
+                model_id: 0,
+                battery: 'low',
+                deviceid: 'withingsDevideId',
+                timezone: 'string',
+                last_session_date: 0,
+              },
+              {
+                type: 'string',
+                model: 'string',
+                model_id: 0,
+                battery: 'no',
+                deviceid: 'withingsDevideId2',
+                timezone: 'string',
+                last_session_date: 0,
+              },
+              {
+                type: 'string',
+                model: 'string',
+                model_id: 0,
+                battery: 'medium',
+                deviceid: 'withingsDevideId3',
+                timezone: 'string',
+                last_session_date: 0,
+              },
+              {
+                type: 'string',
+                model: 'string',
+                model_id: 0,
+                battery: 'high',
+                deviceid: 'withingsDevideId4',
+                timezone: 'string',
+                last_session_date: 0,
+              },
+            ],
+          },
+        }),
+      },
+    });
+
+    server.on({
```

Comment:
I think this is the 3rd time that I see this very big test init.
Extract this in a single, or mock more.
But initialization looks very very complicated.

Reply:
I split this init in separate file